detekcija plave loptice.py

```diff
@@ -0,0 +1,86 @@
+import cv2
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+
+
+cap = cv2.VideoCapture(0)  
+
+
+#kalmanov filter za pracenje loptice
+kalman = cv2.KalmanFilter(4, 2)
+kalman.measurementMatrix = np.array([[1, 0, 0, 0],
+                                     [0, 1, 0, 0]], np.float32)
+kalman.transitionMatrix = np.array([[1, 0, 1, 0],
+                                    [0, 1, 0, 1],
+                                    [0, 0, 1, 0],
+                                    [0, 0, 0, 1]], np.float32)
+kalman.processNoiseCov = np.eye(4, dtype=np.float32) * 0.03
+
+prev_x, prev_y = None, None
+
+while True:
+    ret, frame = cap.read()
+
+    if not ret:
+        break
+
+    hsv = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+
+
+    lower_plava = np.array([100, 150, 50]) #podesava u zavisnosti od boje loptice koja se koristi
+    upper_plava = np.array([140, 255, 255])
+
+
+    #ostavlja samo opseg boja u kojima se nalazi loptica radi preci
+    mask = cv2.inRange(hsv, lower_plava, upper_plava)
+
+    blue_only = cv2.bitwise_and(frame, frame, mask=mask)
+
+    gray = cv2.cvtColor(blue_only, cv2.COLOR_BGR2GRAY)
+
+    blurred = cv2.GaussianBlur(gray, (9, 9), 2)
+
+    circles = cv2.HoughCircles(blurred, cv2.HOUGH_GRADIENT, dp=1.2, minDist=100, param1=100, param2=30, minRadius=10, maxRadius=100)
+
+    if circles is not None:
+        circles = np.round(circles[0, :]).astype("int")
+        for (x, y, r) in circles:
+            cv2.circle(frame, (x, y), r, (0, 255, 0), 4)
+            cv2.rectangle(frame, (x - 5, y - 5), (x + 5, y + 5), (0, 128, 255), -1)
+
+        frame_centar_x = frame.shape[1] // 2
+        frame_centar_y = frame.shape[0] // 2
```

Comment:
Kada budete imali klasu. Dodajte neku metodu koja je "set_grame_center", da se ne bi oslanjali da nam je centar lopte u centru frejma (na taj način možemo da približimo/zumiramo kameru da bi imali bolju preciznost).